pkg/kubecost/summaryallocation.go

```diff
@@ -0,0 +1,1155 @@
+	// populated in step (#) and used to "filter" out the correct proportion
+	// of idle costs, given that non-idle allocations have been filtered.
+	var filteredTotals map[string]*ResourceTotals
+	if len(aggSet.idleKeys) > 0 && len(options.FilterFuncs) > 0 {
+		filteredTotals = make(map[string]*ResourceTotals, len(aggSet.idleKeys))
+
+		// If costs are shared, record those resource totals here so that idle
+		// for the shared resources gets included.
+		for key, rt := range sharedResourceTotals {
+			if _, ok := filteredTotals[key]; !ok {
+				filteredTotals[key] = &ResourceTotals{}
+			}
+
+			// TODO do we need PV, Network, LoadBalancer here?
+
+			filteredTotals[key].CPUCost += rt.CPUCost
+			filteredTotals[key].GPUCost += rt.GPUCost
+			filteredTotals[key].RAMCost += rt.RAMCost
+		}
+	}
+
+	sharingCoeffs := map[string]float64{}
+
+	// (#) Distribute idle cost, if sharing
+	// (#) Distribute tenancy costs, if sharing
+	// (#) Aggregate
+	// TODO better comment
+	for _, sa := range sas.SummaryAllocations {
+		// Generate key to use for aggregation-by-key and allocation name
+		key := sa.generateKey(aggregateBy, options.LabelConfig)
+
+		// TODO comment
+		// Do this before adding idle cost
+		if options.ShareSplit == ShareEven {
+			sharingCoeffs[key] += 1.0
+		} else {
+			sharingCoeffs[key] += sa.TotalCost()
+		}
+
+		// Distribute idle allocations according to the idle coefficients
+		// NOTE: if idle allocation is off (i.e. ShareIdle == ShareNone) then
+		// all idle allocations will be in the aggSet at this point, so idleSet
+		// will be empty and we won't enter this block.
+		if len(idleSet.SummaryAllocations) > 0 {
+			// Distribute idle allocations by coefficient per-idleId, per-allocation
+			for _, idle := range idleSet.SummaryAllocations {
+				var key string
+
+				// Only share idle allocation with current allocation (sa) if
+				// the relevant property matches (i.e. Cluster or Node,
+				// depending on which idle sharing option is selected)
+				if options.IdleByNode {
+					if idle.Properties.Node != sa.Properties.Node {
+						continue
+					}
+
+					key = idle.Properties.Node
+				} else {
+					if idle.Properties.Cluster != sa.Properties.Cluster {
+						continue
+					}
+
+					key = idle.Properties.Cluster
+				}
+
+				cpuCoeff, gpuCoeff, ramCoeff := ComputeIdleCoefficients(options.ShareIdle, key, sa.CPUCost, sa.GPUCost, sa.RAMCost, allocTotals)
+
+				sa.CPUCost += idle.CPUCost * cpuCoeff
+				sa.GPUCost += idle.GPUCost * gpuCoeff
+				sa.RAMCost += idle.RAMCost * ramCoeff
+			}
+		}
+
+		// The key becomes the allocation's name, which is used as the key by
+		// which the allocation is inserted into the set.
+		sa.Name = key
+
+		// If merging unallocated allocations, rename all unallocated
+		// allocations as simply __unallocated__
+		if options.MergeUnallocated && sa.IsUnallocated() {
+			sa.Name = UnallocatedSuffix
+		}
+
+		// TODO do we need to be going off of "cluster/node" for "node" in all these places?
```

Comment:
I think this is the same question as my comment just above. I would guess that the answer is yes, but it won't be a problem in _most_ situations because cloud providers tend to be pretty verbose and unique in the way they name cluster nodes. There is no guarantee, though, especially when thinking about the "long tail" of cluster configurations.

Reply:
Ha yep, this one.